src/main/java/httpserver/HttpParser.java

```diff
@@ -2,12 +2,18 @@
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 
 public class HttpParser implements Parseable{
     @Override
-    public HttpRequest parse(String request) {
+    public Optional<HttpRequest> parse(String request) {
         List<String> splitRequest = Arrays.asList(request.split(System.lineSeparator()));
         List<String> splitRequestLine = Arrays.asList(splitRequest.get(0).split(" "));
-        return new HttpRequest(splitRequestLine.get(0), splitRequestLine.get(1));
+        HttpRequest httpRequest = null;
+        if(splitRequestLine.size() == 3)
+        {
+            httpRequest = new HttpRequest(splitRequestLine.get(0), splitRequestLine.get(1));
+        }
+        return Optional.ofNullable(httpRequest);
```

Comment:
For whatever reason, I try to avoid reassigning variables after I create them. In a situation like this, I would opt for the following
```suggestion
        if(splitRequestLine.size() == 3) {
            String method = splitRequestLine.get(0);
            String path = splitRequestLine.get(1);
            return Optional.of(new HttpRequest(method, path);
        }
        return Optional.empty();
```

Reply:
Also, what does `splitRequestLine.size() == 3` mean? Is there a way we could make this more intention revealing?

Reply:
I'll refactor